src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,4296 @@
+        DesignSpecMSHP::getDesignSpecMSHPdata(errorsFound);
+
+        if (errorsFound) {
+            // show fatal warning
+        }
+    }
+
+    void DesignSpecMSHP::getDesignSpecMSHPdata(bool errorsFound)
+    {
+        std::string cCurrentModuleObject = "UnitarySystemPerformance:Multispeed";
+
+        auto const instances = inputProcessor->epJSON.find(cCurrentModuleObject);
+        if (instances == inputProcessor->epJSON.end()) {
+            errorsFound = true;
+        } else {
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = instance.key();
+                DesignSpecMSHP thisDesignSpec;
+
+                thisDesignSpec.name = thisObjectName;
+                thisDesignSpec.numOfSpeedHeating = fields.at("number_of_speeds_for_heating"); // required field
+                thisDesignSpec.numOfSpeedCooling = fields.at("number_of_speeds_for_cooling"); // required field
+                int maxSpeeds = max(thisDesignSpec.numOfSpeedHeating, thisDesignSpec.numOfSpeedCooling);
+                thisDesignSpec.designSpecMSHPType_Num = 1; // add global int value for factory
+
+                std::string loc_singleModeOp("No");
+                if (fields.find("single_mode_operation") != fields.end()) { // not required field
+                    loc_singleModeOp = UtilityRoutines::MakeUPPERCase(fields.at("single_mode_operation"));
+                }
+                // set JSON data to design spec object
+                if (UtilityRoutines::SameString(loc_singleModeOp, "Yes")) {
+                    thisDesignSpec.singleModeFlag = true;
+                } else if (UtilityRoutines::SameString(loc_singleModeOp, "No")) {
+                    thisDesignSpec.singleModeFlag = false;
+                } else {
+                }
+
+                Real64 loc_noLoadAirFlowRateRatio(1.0);
+                if (fields.find("no_load_supply_air_flow_rate_ratio") != fields.end()) { // not required field
+                    loc_noLoadAirFlowRateRatio = fields.at("no_load_supply_air_flow_rate_ratio");
+                }
+                thisDesignSpec.noLoadAirFlowRateRatio = loc_noLoadAirFlowRateRatio;
+
+                thisDesignSpec.MSHeatingSpeedRatio.resize(maxSpeeds);
+                thisDesignSpec.MSCoolingSpeedRatio.resize(maxSpeeds);
+
+                auto speedFlowRatios = fields.find("flow_ratios");
+                if (speedFlowRatios != fields.end()) {
+                    auto flowRatioArray = speedFlowRatios.value();
+                    int speedNum = -1;
+                    for (auto flowRatio : flowRatioArray) {
+                        speedNum += 1;
+                        auto coolingSpeedRatioObject = flowRatio.at("cooling_speed_supply_air_flow_ratio");
+                        if (coolingSpeedRatioObject == "Autosize") {
```

Comment:
You shouldn't need to make it case insensitive. If the field is Autosizable then it should always be "Autosize" since the idf->epJSON does this automatically and the schema validation will throw an error if it isn't "Autosize".

This way is the appropriate method for right now, however, there might be a better way to check if a field is autosized/autocalculated. I need to spend a little bit of time thinking through a better approach (likely a function within InputProcessor).